http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -823,6 +786,142 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
+    ev: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- ev.ask
+      diskResourceOpt <- id match {
+        case Left(runtimeId) =>
+          controlledResourceQuery
+            .getWsmRecordForRuntime(runtimeId, WsmResourceType.AzureDisk)
+            .transaction
+        case Right(diskResourceId) =>
+          controlledResourceQuery
+            .getWsmRecordFromResourceId(diskResourceId, WsmResourceType.AzureDisk)
+            .transaction
+      }
+      diskResource <- F.fromOption(
+        diskResourceOpt,
+        AzureDiskResourceDeletionError(id,
+                                       workspaceId,
+                                       "No disk resource found for delete azure disk. No-op for wsmDao.deleteDisk."
+        )
+      )
+      jobId = getWsmJobId("delete-disk", diskResource.resourceId)
+
+      _ <- diskResourceOpt.traverse { disk =>
+        for {
+          _ <- logger.info(ctx.loggingCtx)(s"Sending WSM delete message for disk resource ${disk.resourceId.value}")
+          _ <- wsmDao
+            .deleteDisk(
+              DeleteWsmResourceRequest(
+                workspaceId,
+                disk.resourceId,
+                DeleteControlledAzureResourceRequest(
+                  WsmJobControl(jobId)
+                )
+              ),
+              auth
+            )
+            .void
+            .adaptError(e =>
+              AzureDiskDeletionError(
+                disk.resourceId,
+                workspaceId,
+                s"${ctx.traceId.asString} | WSM call to delete disk failed due to ${e.getMessage}. Please retry delete again"
+              )
+            )
+        } yield ()
+      }.void
+    } yield ()
+
+  override def deleteDisk(msg: DeleteDiskV2Message)(implicit ev: Ask[F, AppContext]): F[Unit] = {
+    implicit val wsmDeleteDiskDoneCheckable: DoneCheckable[GetDeleteJobResult] = (v: GetDeleteJobResult) =>
+      v.jobReport.status.equals(WsmJobStatus.Succeeded) || v.jobReport.status == WsmJobStatus.Failed
+    for {
+      ctx <- ev.ask
+      auth <- samDAO.getLeoAuthToken
+
+      wsmResourceId <- F.fromOption(
+        msg.wsmResourceId,
+        DiskDeletionError(
+          msg.diskId,
+          msg.workspaceId,
+          s"No associated WsmResourceId found for Azure disk"
+        )
+      )
+
+      _ <- deleteDiskResource(Right(wsmResourceId), msg.workspaceId, auth)
+
+      deleteJobId = getWsmJobId("delete-disk", wsmResourceId)
+      getDeleteJobResult = wsmDao.getDeleteDiskJobResult(
+        GetJobResultRequest(msg.workspaceId, deleteJobId),
+        auth
+      )
+
+      taskToRun = for {
+        // We need to wait until WSM deletion job to be done to update the database
+        resp <- streamFUntilDone(
+          getDeleteJobResult,
+          config.deleteDiskPollConfig.maxAttempts,
+          config.deleteDiskPollConfig.interval
+        ).compile.lastOrError
+
+        _ <- resp.jobReport.status match {
+          case WsmJobStatus.Succeeded =>
```

Comment:
feels like this pattern of 'perform x,y,z based on exhasutive match of WsmJobStatus` should be encapsulated elsewhere, but a bit out of scope here.